frontend/packages/dev-console/src/components/import/image-search/ImageStream.tsx

```diff
@@ -1,192 +1,187 @@
         </div>
         <div className="col-lg-4 col-md-4 col-sm-4 col-xs-12">
           <DropdownField
             name="imageStream.tag"
             label="Tag"
-            items={getImageStreamTags(values.imageStream.image)}
-            title="Tag"
-            disabled={values.imageStream.image === ''}
+            items={imageStreamTagList}
+            title={
+              isNamespaceSelected && isImageStreamSelected && !isTagsAvailable
+                ? 'No Tag'
+                : 'Select Tag'
+            }
+            disabled={!isImageStreamSelected}
             fullWidth
             required
+            onChange={(tag) => {
+              tag !== '' && searchImageTag(tag);
+            }}
           />
           <div className="odc-imagestream-separator">:</div>
         </div>
       </div>
-      {values.registry === registryType.Internal &&
-        imageNamespace.value &&
-        imageNamespace.value !== builderImagesNamespace.Openshift &&
-        values.project.name !== imageNamespace.value && (
-          <div className="row odc-imagestream-alert">
-            <div className="col-lg-12">
-              <Alert
-                variant="warning"
-                title="add cluster policy to use internal imagestream"
-                isInline
-              >
-                Service account <strong>default</strong> will need image pull authority to deploy
-                images from <strong>{imageNamespace.value}</strong>. You can grant authority with
-                the command:
-                <p>
-                  <code>
-                    oc policy add-role-to-user system:image-puller system:serviceaccount:
-                    {values.project.name}:default -n {imageNamespace.value}
-                  </code>
-                </p>
-              </Alert>
-            </div>
-          </div>
-        )}
+      {isNamespaceSelected && isImageStreamSelected && !isTagsAvailable && hasCreateAccess && (
+        <div className="col-lg-12 odc-imagestream-alert">
+          <Alert variant="warning" title="No Image streams tags found" isInline>
+            No tags are available in image stream {imageStream.image}
+          </Alert>
+        </div>
+      )}
+      {isNamespaceSelected && !isStreamsAvailable && hasCreateAccess && (
+        <div className="col-lg-12 odc-imagestream-alert">
+          <Alert variant="warning" title="No Image streams found" isInline>
+            No image streams are available in project {imageStream.namespace}
+          </Alert>
+        </div>
+      )}
+      {isNamespaceSelected && !hasCreateAccess && (
+        <div className="col-lg-12 odc-imagestream-alert">
+          <Alert variant="warning" title="Permission denied" isInline>
+            Service account default does not have authority to pull images from{' '}
+            {imageStream.namespace}. Select another project to continue.
+          </Alert>
+        </div>
```

Comment:
@karthikjeeyar I see that in alert notifications only title and message is different for each alert so I think it can be converted into a small generic FC where we simply pass the title and message. Also in that way it can be used outside this file if in future we want to display similar kind of notification. What do you think?

Reply:
Even if we create a new component we need to pass the same arguments title and message to it like we do in Alert component. I think its an overkill to have component wrapper for Alert component here. your thoughts?

Reply:
Yeah, probably I am thinking too much on this :P